File: app/src/main/java/com/naveed/samples/helper/base/SplashBase.java
package com.naveed.samples.helper.base;

import android.os.Bundle;
import android.os.Handler;
import android.support.annotation.Nullable;

/**
 * Created by naveedali on 10/8/17.
 */

public abstract class SplashBase extends BaseActivity {

    private final static int TIME_DELAY = 2500;
    private Class<?> activityClass;
    Handler mHandler;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mHandler = new Handler();

    }

    protected void startDelay() {
        mHandler.postDelayed(runnable, TIME_DELAY);
    }

    Runnable runnable = new Runnable() {
        @Override
        public void run() {
            onDelayComplete();
        }
    };

    protected abstract void onDelayComplete();

}

File: app-mvp/src/main/java/com/naveed/samples/helper/base/SplashBase.kt
package com.naveed.samples.helper.base

import android.os.Bundle
import android.os.Handler

/**
 * Created by naveedali on 10/8/17.
 */

abstract class SplashBase : BaseActivity() {
    private val activityClass: Class<*>? = null
    private lateinit var mHandler: Handler

    private var runnable: Runnable = Runnable { onDelayComplete() }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mHandler = Handler()

    }

    protected fun startDelay() {
        mHandler.postDelayed(runnable, TIME_DELAY.toLong())
    }

    protected abstract fun onDelayComplete()

    companion object {
        private const val TIME_DELAY = 2500
    }

}
